src/service.ts

```diff
@@ -256,12 +265,18 @@ export class DebugService implements IDebugger {
     });
   };
 
-  private onContinued() {
+  private clearCallStack() {
     this._model.linesCleared.emit();
     this._model.callstackModel.frames = [];
     this._model.variablesModel.scopes = [];
   }
 
+  private clearModel() {
+    this.clearCallStack();
+    this._model.breakpointsModel.breakpoints = [];
```

Comment:
The breakpoints are removed from the model but not (yet) in the UI?

Reply:
Fix incoming ;)

Reply:
* Doesn't setting that `[]` automatically trigger the `breakpointsChanged` signal?
* Do we needo *both* `breakpointChanged` and also `breakpointsChanged`? I think if all the models just had a simple `changed` value and they re-rendered it might be okay, because it would be the virtual DOM's responsibility to re-render efficiently, maybe. What do you think?

Reply:
My understanding of the workflow is the following:
- when clicking on the gutter to set a breakpoint, the setBreakpoints method of the model is called
- this last one triggers a `breakpointsChanged` signal
- the cell handler listens to this signal and calls the `updateBreakpoints` on the service object, which sends a debug request to the kernel to set the new breakpoints.

What we want to do here is the opposite: updating the gutter based on changes of the model.

Regarding the signals I don't know what is the most efficient option, but I agree that it would be better if we could keep only one signal.

Reply:
Do you think it would be better not to set the gutter markers in `onGutterClick` but rather when the cell handler receives a `breakpointsChanged` signal?

Reply:
> Doesn't setting that `[]` automatically trigger the `breakpointsChanged` signal?

I think that's the case yes.

Agree there should be only one `breakpointsChanged :+1:

Reply:
Actually I wonder if we should not update the gutter based on the breakpointsChanged event only, and adopt this behavior for other parts of the UI:
- a user's input calls some methods of the service, but we do not update parts of the view that depends on data of the model.
- the service sends a request to the backend if that's needed, and updates the mdel
- updating the model triggers a signal which forces the UI to rerender.